lib/Geo/GeoJson.cpp

```diff
@@ -39,616 +41,750 @@
 #include "Basics/VelocyPackHelper.h"
 #include "Basics/debugging.h"
 #include "Geo/GeoParams.h"
+#include "Geo/ShapeContainer.h"
 #include "Geo/S2/S2MultiPointRegion.h"
 #include "Geo/S2/S2MultiPolyline.h"
-#include "Geo/ShapeContainer.h"
 #include "Logger/Logger.h"
 
+namespace arangodb::geo::json {
 namespace {
-// Have one of these values:
-std::string const kTypeStringPoint = "Point";
-std::string const kTypeStringLineString = "LineString";
-std::string const kTypeStringPolygon = "Polygon";
-std::string const kTypeStringMultiPoint = "MultiPoint";
-std::string const kTypeStringMultiLineString = "MultiLineString";
-std::string const kTypeStringMultiPolygon = "MultiPolygon";
-std::string const kTypeStringGeometryCollection = "GeometryCollection";
-}  // namespace
 
-namespace {
-inline bool sameCharIgnoringCase(char a, char b) noexcept {
-  return arangodb::basics::StringUtils::toupper(a) ==
-         arangodb::basics::StringUtils::toupper(b);
+// The main idea is check in our CI 3 compile time branches:
+// 1. Validation true
+// 2. Validation false kIsMaintainer true
+// 3. Validation false kIsMaintainer false
```

Comment:
```suggestion
// 3. Validation false kIsMaintainer false
// Therefore, we run tests with validation always on on x86_64 and with no validation on aarch64
// to get good test coverage for both cases.
```

Reply:
I will do it in next PR! this branch: https://github.com/arangodb/arangodb/tree/update-geo-s2-analyzer